CompositionRoot/B.swift

```diff
@@ -1,17 +1,17 @@
-protocol BPrototol {
+protocol BProtocol {
     func bMethod()
 }
 
-class B: BPrototol {
+class B: BProtocol {
     
-    private let applicationContext: ApplicationContext
+    private let cFactory: CFactoryProtocol
     
-    init(applicationContext: ApplicationContext) {
-        self.applicationContext = applicationContext
+    init(cFactory: CFactoryProtocol) {
+        self.cFactory = cFactory
     }
     
     func bMethod() {
-        let c = C(applicationContext: applicationContext)
```

Comment:
Hard code dependency on `C`. That make B depend on concrete class `C`, not abstract interface.

Reply:
It's hard to replace `C` with `FakeC` to unit test this function.